content/article/deploying-dancer-apps-addendum.md

```diff
@@ -0,0 +1,164 @@
+  {
+    "title"       : "Deploying Dancer Apps (Addendum)",
+    "authors"     : ["dave-cross"],
+    "date"        : "2024-08-06T16:49:00",
+    "tags"        : ["devops", "deployment", "dancer", "psgi"],
+    "draft"       : false,
+    "image"       : "/images/deploying-dancer-apps/deploying-dancer-apps.png",
+    "thumbnail"   : "/images/deploying-dancer-apps/thumb-deploying-dancer-apps.png",
+    "description" : "Some more thoughts about deploying Dancer apps as persistent daemons",
+    "categories"  : "tooling"
+  }
+
+*This article was originally published at
+[Perl Hacks](https://perlhacks.com/2024/08/deploying-dancer-apps-addendum/).*
+
+---
+
+Back in May, I wrote a blog post about how [I had moved a number of Dancer2
+applications](https://www.perl.com/article/deploying-dancer-apps/) to a new
+server and had, in the process, created a standardised procedure for
+deploying Dancer2 apps. It’s been about six weeks since I did that and I
+thought it would be useful to give a little update on how it all went and
+talk about a few little changes I’ve made.
+
+I mentioned that I was moving the apps to a new server. What I didn’t say was
+that I was convinced my old server was overpowered (and overpriced!) for what
+I needed, so the new server has less RAM and, I think, a slower CPU than the
+old one. And that turned out to be a bit of a problem. It turned out there
+was a time early each morning when there were too many requests coming into
+the server and it ran out of memory. I was waking up most days to a dead
+server. My previous work meant that fixing it wasn’t hard, but it really
+wasn’t something that I wanted to do each most mornings.
+
+So I wanted to look into reducing the amount of memory used by the apps. And
+that turned out to be a two-stage approach.
+
+You might recall that the apps were all controlled using a standardised
+driver program called “app_service”. It looked like this:
+
+```perl
+#!/usr/bin/env perl
+ 
+use warnings;
+use strict;
+use Daemon::Control;
```

Comment:
```suggestion
use Daemon::Control ();
```

Reply:
I'm not going to change that. Not because it's a bad idea, but because it's a change from the code that was published in  the previous article - so the change is just a distraction.